app/services/checksum_validator.rb

```diff
@@ -22,6 +31,7 @@ def initialize(preserved_copy, endpoint_name)
     @preserved_copy = preserved_copy
     @bare_druid = preserved_copy.preserved_object.druid
     @endpoint = Endpoint.find_by(endpoint_name: endpoint_name)
+    valid?
```

Comment:
If this is important mid-initialization, shouldn't it `raise` if invalid?  It seems counterproductive to front-load the validity check and then do nothing with the result.  `endpoint` is not assignable by the caller, so they have no possibility of putting the object into a valid state.  Seems likely they shouldn't have the object at all.

Alternatively, check validity in an `after_initialize` callback, and raise there if necessary.  

The way we have to lookup `Endpoint` by name **per PC** is a performance waste.  Callers should just pass the `Endpoint` themselves, which gives `initialize` the chance to interrogate it for correctness.  Even if we don't subsume `EndpointType` (and we should!), we should add a `Endpoint#archive_type?` that caches the answer, so that the combination of passing the `Endpoint` and calling that gives us:
- the validation we want
- saves millions of `find_by` queries

Reply:
Upon reflection, the method isn't even necessary if we pass the Endpoint object, since the associated EndpointType object will be cached with it.